### PR TITLE
Moving Perf tests to their own csprojs

### PR DESCRIPTION
**Pull request by dsgouda**
- Moved Perf tests to their own csproj files
- Removed perf test references from functional test projs
- Moved project.jsons under System.Runtime.Serialization.Xml, System.Xml.XmlSerializer.Tests and - System.Runtime.Serialization.Json to ContractReferences to work with UseContractReferences flag
- Added perf test csprojs to related .sln files 
- Generating separate .Performance.Tests.dll for perf tests

This cleanup should help with discovery and execution of perf tests using latest changes in build tools
@shmao PTAL System.Runtime.Serialization.* changes

@karajas @MattGal @lt72 @ericstj @mellinoe

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/tests/SampleClasses.cs [hunk 1]
```diff
@@ -82,7 +82,13 @@ public MyTypeListConverter(Type[] types)
         }
     }
 
+#if FUNCTIONAL_TESTS
     [TypeConverter("System.ComponentModel.Tests.BaseClassConverter, System.ComponentModel.TypeConverter.Tests, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null")]
+#elif PERFORMANCE_TESTS
```
Comment (karajas) [inline src/System.ComponentModel.TypeConverter/tests/SampleClasses.cs]: This is the only change to test code to allow compilation for System.ComponentModel.TypeConverter

File: src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <OutputType>Library</OutputType>
+    <RootNamespace>System.Collections.NonGeneric.Performance.Tests</RootNamespace>
+    <AssemblyName>System.Collections.NonGeneric.Performance.Tests</AssemblyName>
+    <ProjectGuid>{EE95AE39-845A-42D3-86D0-8065DBE56612}</ProjectGuid>
```
Comment (mellinoe) [inline src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj]: This is the same GUID as the unit test project, probably a copy-paste issue.

While I'm commenting on this one, I'd recommend just removing the following properties from all of the new projects you created. I know they are probably also in the unit test projects and got copied from there, but you can just leave the existing projects alone and only clean them from the new projects.

* Configuration
* Platform
* OutputType
* ProjectGuid (VS might add this back in some cases, but just delete it to start out I think).

File: src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <IncludePerformanceTests>true</IncludePerformanceTests>
+    <NugetTargetMoniker>.NETStandard,Version=v1.3</NugetTargetMoniker>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the configurations -->
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+  </PropertyGroup>
```
Comment (mellinoe) [inline src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj]: nit: You can just close this tag on the previous line. Same goes elsewhere in some of the new project files.

File: src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj (hunk 2 above)
Comment (mellinoe) [inline src/System.Collections.NonGeneric/tests/Performance/System.Collections.NonGeneric.Performance.Tests.csproj]: If VS doesn't cooperate with removing the ProjectGuid, just make sure it's not the same as the unit test project GUID.

**Comment by mellinoe** [comment]
I skimmed over the new project files and made a couple clean-up / formatting suggestions. Everything else looks good, as long as everything plays nicely with our build scripts, and all of the unit tests, etc. are still executed as before.

File: src/System.Runtime.Serialization.Json/tests/Performance/System.Runtime.Serialization.Json.Performance.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>System.Runtime.Serialization.Json.Performance.Tests</RootNamespace>
+    <AssemblyName>System.Runtime.Serialization.Json.Performance.Tests</AssemblyName>
+    <ProjectGuid>{701CB3BC-00DC-435D-BDE4-C5FC29A708A7}</ProjectGuid>
+    <NugetTargetMoniker>.NETStandard,Version=v1.3</NugetTargetMoniker>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(UseContractReferences)' == 'true'" >
```
Comment (shmao) [inline src/System.Runtime.Serialization.Json/tests/Performance/System.Runtime.Serialization.Json.Performance.Tests.csproj]: @dsgouda can you please confirm if this test project still builds when UseContractReferences==true?

Comment (dsgouda) [inline src/System.Runtime.Serialization.Json/tests/Performance/System.Runtime.Serialization.Json.Performance.Tests.csproj]: @shmao Yes I was able to build and test using the UserContractReferences=true condition

Comment (shmao) [inline src/System.Runtime.Serialization.Json/tests/Performance/System.Runtime.Serialization.Json.Performance.Tests.csproj]: Thanks for the ACK. The changes made to System.Runtime.Serialization.* LGTM.

**Comment by mellinoe** [comment]
Cleanup looks good to me.

**Comment by mellinoe** [comment]
@dsgouda Could you rebase this onto master (to remove the explicit merge commit)?

**Comment by lt72** [comment]
LGTM

**Pull request merged**